src/main/java/org/cryptomator/cryptofs/health/dirid/OrphanDir.java

```diff
@@ -33,5 +58,119 @@ public String toString() {
 		return Map.of(ENCRYPTED_PATH, dir.toString());
 	}
 
-	// fix: create new dirId inside of L+F dir and rename existing dir accordingly.
+	@Override
+	public void fix(Path pathToVault, VaultConfig config, Masterkey masterkey, Cryptor cryptor) throws IOException {
+		var sha1 = getSha1MessageDigest();
+		String runId = BaseEncoding.base64Url().encode(sha1.digest(UUID.randomUUID().toString().getBytes(StandardCharsets.UTF_8))).substring(0, 3);
+		Path orphanedDir = pathToVault.resolve(Constants.DATA_DIR_NAME).resolve(this.dir);
+		String orphanHash = dir.getParent().getFileName().toString() + dir.getFileName().toString();
+
+		var stepParentDir = prepareCryptoFilesystem(pathToVault, cryptor.fileNameCryptor(), orphanHash);
+
+		try (var orphanedContentStream = Files.newDirectoryStream(orphanedDir)) {
+			AtomicInteger fileCounter = new AtomicInteger(1);
+			AtomicInteger dirCounter = new AtomicInteger(1);
+			AtomicInteger symlinkCounter = new AtomicInteger(1);
+			String longNameSuffix = createClearnameToBeShortened(config.getShorteningThreshold());
+
+			for (Path orphanedResource : orphanedContentStream) {
+				var newClearName = switch (determineCiphertextFileType(orphanedResource)) {
+					case FILE -> FILE_PREFIX + fileCounter.getAndIncrement();
+					case DIRECTORY -> DIR_PREFIX + dirCounter.getAndIncrement();
+					case SYMLINK -> SYMLINK_PREFIX + symlinkCounter.getAndIncrement();
+				} + "_" + runId;
+				adoptOrphanedResource(new Adoption(newClearName, orphanedResource), stepParentDir, cryptor.fileNameCryptor(), longNameSuffix, sha1);
+			}
+		}
+		Files.delete(orphanedDir);
+	}
+
+	// visible for testing
+	CryptoPathMapper.CiphertextDirectory prepareCryptoFilesystem(Path pathToVault, FileNameCryptor cryptor, String clearStepParentDirName) throws IOException {
```

Comment:
```suggestion
	CryptoPathMapper.CiphertextDirectory prepareStepParent(Path pathToVault, FileNameCryptor cryptor, String clearStepParentDirName) throws IOException {
```

Reply:
Rename applies to test methods and their display names as well

Reply:
superseeded with 420e200